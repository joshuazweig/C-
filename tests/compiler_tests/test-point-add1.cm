int main() {
    stone a;
    stone b;
    stone c;
    stone d;

    stone a1;
    stone b1;
    stone p;

    a = "25";
    b = "37";
    c = "19";
    d = "8";

    a1 = "7";
    b1 = "26";
    p = "61";

    mint A;
    mint B;
    curve *E;

    A = <a1, p>;
    B = <b1, p>;
    E = <A, B>;

    point *P;
    point *Q;

    P = <E, a, b>;
    Q = <E, c, d>;

    print_point(P+Q); // should print (59, 2)
    return 0;
}
