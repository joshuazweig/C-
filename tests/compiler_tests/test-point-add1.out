a: 7
b: 26
p: 61
<25, 37>
<19, 8>
<59, 2>
<58, 51>
<27, 16>
